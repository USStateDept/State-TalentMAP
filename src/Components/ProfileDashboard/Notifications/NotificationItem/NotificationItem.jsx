import PropTypes from 'prop-types';
import InformationDataPoint from '../../InformationDataPoint';
import { getTimeDistanceInWords } from '../../../../utilities';

const NotificationItem = ({ content, notificationTime }) => {
  const timeDistanceInWords = getTimeDistanceInWords(notificationTime);
  return (
    <div className="usa-grid-full notification-card">
      <InformationDataPoint content={content} title={timeDistanceInWords} titleOnBottom />
    </div>
  );
};

NotificationItem.propTypes = {
  content: PropTypes.node.isRequired,
  notificationTime: PropTypes.string.isRequired,
};

export default NotificationItem;
