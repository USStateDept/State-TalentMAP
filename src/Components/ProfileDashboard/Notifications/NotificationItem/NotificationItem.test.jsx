import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { subMonths } from 'date-fns';
import NotificationItem from './NotificationItem';

describe('NotificationItemComponent', () => {
  it('is defined', () => {
    const wrapper = shallow(
      <NotificationItem
        content="content"
        notificationTime="10-10-2017"
      />,
    );
    expect(wrapper).toBeDefined();
  });

  xit('matches snapshot', () => {
    // Use subMonths so that snapshot doesn't go out of date every month.
    // Using "2" means that our snapshot will some times render "about 2 months ago"
    // and some times "2 months ago" as its title, since it's technically exact.
    // We subtract 2.1 so that it's not exactly 2 months ago, which
    // should always render "about 2 months ago".
    // https://date-fns.org/v1.29.0/docs/distanceInWords
    const notificationTime = subMonths(new Date(), 2.1).toString();
    const wrapper = shallow(
      <NotificationItem
        content="content"
        notificationTime={notificationTime}
      />,
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
