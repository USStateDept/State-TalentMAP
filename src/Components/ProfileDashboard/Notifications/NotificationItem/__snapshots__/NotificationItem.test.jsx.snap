// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationItemComponent matches snapshot 1`] = `
<div
  className="usa-grid-full notification-card"
>
  <InformationDataPoint
    className=""
    content="content"
    sideBySide={false}
    title="about 2 months ago"
    titleOnBottom={true}
  />
</div>
`;
